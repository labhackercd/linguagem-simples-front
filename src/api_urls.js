module.exports = Object.freeze({

    //NAME CONSTANTS
    APPLICATION_NAME: "Estudio",

    //APPLICATION URL'S
        APPLICATION_FRONT_BASE_URL: "www.teste.com",  //APPLICATION FRONT BASE URL
        //APPLICATION_SERVER_API_BASE_URL: 'http://localhost:8000/api/', //APPLICATION API SERVER BASE URL
        APPLICATION_SERVER_API_BASE_URL: 'https://acompanhe.camara.leg.br/server/api/',
        APPLICATION_RESET_PASSWORD_URL: 'https://acompanhe.camara.leg.br/server/password_reset/',

    //ROUTES
        INITIAL_PAGE_URL: '/', //Application Login and Initial page
        ESTUDIO_PAGE_URL: '/estudio', //Application Estudio Page
        DASHBOARD_PAGE_URL: '/dashboard/:dashboardId', //Application Dashboard page
        DASHBOARD_BASE_URL: '/dashboard/',

    //API'S CAMARA DOS DEPUTADOS
        API_CD_EVENTOS_PLENARIO: 'https://dadosabertos.camara.leg.br/api/v2/eventos?codTipoEvento=110',// 110 code corresponds to Plenary
        API_CD_EVENTOS: 'https://dadosabertos.camara.leg.br/api/v2/eventos/',
        API_CD_ORIENTACAO_VOTACAO: 'https://liderancadigital.camara.leg.br/proxy_pass/secod-wsVotDecom-orientacao',
        API_CD_ITENS_EM_VOTACAO: 'https://infoleg.camara.gov.br/wsVotDecom/votacao/itens-em-votacao-na-reuniao/',

    //API'S BACKEND
        //AUTH URL'S
            TOKEN_REFRESH_URL: '/token/refresh/', //Application Api Token REFRESH PATH
            TOKEN_OBTAIN_URL: '/token/obtain/', //Application Api Token PATH
            TOKEN_VERIFY_URL: '/token/verify/', //Application Api Token Verify

        //FILE-VIDEO
            API_FILE_VIDEO_URL:'/file-video/', //Application end-point to obtain the source of specific video

        //NEWS
            API_NEWS_URL:'/newsâ€‹/news_list', //Obtain news list
            API_NEWS_SEARCH_URL:'/news-search/', //Obtain specific news using parameters

        //PUBLICATIONS
            API_PUBLICATIONS_URL:'/publications/',

        //RADIO AGENCY
            API_RADIO_AGENCY_URL:'/radioagency/',
            API_RADIO_AGENCY_SEARCH_URL:'/radioagency-search/',

        //RADIO CAMARA
            API_RADIO_CAMARA_URL:'/radiocamara/',
            API_RADIO_CAMARA_SEARCH_URL:'/radiocamara-search/',

        //SAVED CONTENTS
            API_SAVED_CONTENTS_URL:'/saved-contents/',

        //SESSIONS
            API_SESSIONS_URL:'/sessions/',

        //TV CAMARA
            API_TV_CAMARA_URL:'/tvcamara/',
            API_TV_CAMARA_SEARCH_URL:'/tvcamara-search/',

        //VIDEOS
            API_VIDEOS_SNIPPETS:'/videos-session/'
  });
