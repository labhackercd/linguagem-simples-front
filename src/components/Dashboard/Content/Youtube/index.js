import React from 'react';
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography' 

import getYoutubeVideoUrl from './fetchYoutubeVideoUrl'

export default class Youtube extends React.Component {
  constructor(props){
    super(props);
    this.state = { 
        youtubeVideoId: null,
        sessionIdDadosAbertos: this.props.sessionIdDadosAbertos
    };
  }
  

  fetchYoutubeVideo = async term => {

    var data = null;
    //console.log(this.state.sessionIdDadosAbertos)
    if(this.state.sessionIdDadosAbertos !== undefined){
      //console.log(this.state.sessionIdDadosAbertos)
      data = await getYoutubeVideoUrl(this.state.sessionIdDadosAbertos);
      //This split is necessary because we need just the ID of video, since we need 
      //to use the tag embed in the url instead of the watch returned at API
      this.setState({youtubeVideoId:(data.split('?v=').pop())}) 
      this.setState({dataLoaded:true});
    }else{
      //Nothing to do
      //console.log("entrou no loop")
    }


  };

  componentDidMount(){
    this._isMounted = true;
    if(this._isMounted){
        console.log(this.props)
        this.fetchYoutubeVideo();
    }
  }

  render(){
    if(this.props.sessionIdDadosAbertos===null || this.state.youtubeVideoId === null){
      return (
        <Box width={1} height={1}>
          <Box display="flex" justifyContent="center" alignItems="center" width={"100%"} height={"100%"}>
            <Typography variant="h5" style={{color: "#C4C4C4"}}>Transmissão não disponível</Typography>
          </Box>
        </Box>
      )
    }

    return (
      <div
        className="video"
        id={"transmissaoYoutube"+this.state.youtubeVideoId}
        style={{
          position: "relative",
          paddingBottom: "56.25%" /* 16:9 */,
          paddingTop: 0,
          height: 0
        }}
      >
        <iframe
          style={{
            position: "absolute",
            top: 0,
            left: 0,
            width: "100%",
            height: "100%"
          }}
          src={`https://www.youtube.com/embed/${this.state.youtubeVideoId}`}
          frameBorder="0"
          title="Trasmissão da Sessão Plenária"
        />
      </div>
    );
  }
}
