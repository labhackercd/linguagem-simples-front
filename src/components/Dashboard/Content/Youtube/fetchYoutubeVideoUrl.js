import axios from 'axios' 
import {API_CD_EVENTOS} from '../../../../api_urls'

export default async function getYoutubeVideoUrl(sessionIdDadosAbertos){
    const response = await axios.get(API_CD_EVENTOS+sessionIdDadosAbertos);

    return response.data.dados.urlRegistro;
}