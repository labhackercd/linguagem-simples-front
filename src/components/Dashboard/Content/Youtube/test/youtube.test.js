import Youtube from '../index'
import React from 'react';
import {shallow} from "enzyme/build";
import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import axios from 'axios'
import getYoutubeVideoUrl from '../fetchYoutubeVideoUrl'
import {API_CD_EVENTOS} from '../../../../../api_urls'


test("Test if youtube returns transmission not available when sessionID is null", async (done) => {
    const component = mount(<Youtube sessionIdDadosAbertos={null}></Youtube>);
    //console.log(component.debug())
    //console.log(component.text())
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch(/Transmissão não disponível/i)

    done()
});

describe('Test if youtube returns as expected when correct information are provided', () => {
    //var mock = new MockAdapter(axios);
    const sessionMockData = {
          "dados": {
            "uriDeputados": null,
            "uriConvidados": null,
            "fases": null,
            "id": 59895,
            "uri": "https://dadosabertos.camara.leg.br/api/v2/eventos/59895",
            "dataHoraInicio": "2020-09-01T11:00",
            "dataHoraFim": "2020-09-01T14:30",
            "situacao": "Encerrada",
            "descricaoTipo": "Sessão Deliberativa",
            "descricao": "Sessão Deliberativa Extraordinária (virtual)\r\n",
            "localExterno": null,
            "orgaos": [
              {
                "id": 180,
                "uri": "https://dadosabertos.camara.leg.br/api/v2/orgaos/180",
                "sigla": "PLEN",
                "nome": "Plenário",
                "apelido": "Plenário",
                "codTipoOrgao": 26,
                "tipoOrgao": "Plenário Virtual"
              }
            ],
            "requerimentos": [],
            "localCamara": {
              "nome": "Plenário da Câmara dos Deputados",
              "predio": null,
              "sala": null,
              "andar": null
            },
            "urlDocumentoPauta": "https://dadosabertos.camara.leg.br/api/v2/eventos/59895/pauta",
            "urlRegistro": "https://www.youtube.com/watch?v=DlNbwp9U0wI"
          },
          "links": [
            {
              "rel": "self",
              "href": "https://dadosabertos.camara.leg.br/api/v2/eventos/59895"
            },
            {
              "rel": "related",
              "href": "https://www.youtube.com/watch?v=DlNbwp9U0wI",
              "type": "text/html"
            }
          ]
        ,
        "status": 200,
        "statusText": "",
        "headers": {
          "cache-control": "private",
          "content-type": "application/json;charset=UTF-8",
          "link": "<https://dadosabertos.camara.leg.br/api/v2/eventos/59895>; rel=\"self\",<https://www.youtube.com/watch?v=DlNbwp9U0wI>; rel=\"related\",",
          "x-total-count": "1"
        },
        "config": {
          "url": "https://dadosabertos.camara.leg.br/api/v2/eventos/59895",
          "method": "get",
          "headers": {
            "Accept": "application/json, text/plain, */*"
          },
          "transformRequest": [
            null
          ],
          "transformResponse": [
            null
          ],
          "timeout": 0,
          "xsrfCookieName": "XSRF-TOKEN",
          "xsrfHeaderName": "X-XSRF-TOKEN",
          "maxContentLength": -1
        },
        "request": {}
      }

    test("Test if FetchData returns data when  has worked correct", async (done) => {
        var data = null;
        var sessionIdDadosAbertos = 1
        var mock = new MockAdapter(axios);
        
        await mock.onGet(API_CD_EVENTOS+sessionIdDadosAbertos).replyOnce(200,sessionMockData);

        data = await getYoutubeVideoUrl(sessionIdDadosAbertos);
        //console.log(data)
        expect(data).not.toBeNull();
        expect(data).not.toBeUndefined();
        mock.restore();
        done();
    });

    test("Test Youtube video content lifeclycle with data been loaded", async (done) => {

      var sessionIdDadosAbertos = 1;
      var mock = new MockAdapter(axios);
      await mock.onGet(API_CD_EVENTOS+sessionIdDadosAbertos).replyOnce(200,sessionMockData)
              
      const wrapper = await mount(<Youtube sessionIdDadosAbertos={sessionIdDadosAbertos}/>);

      setImmediate(() => {
        wrapper.update();

        const videoDiv = wrapper.find("#transmissaoYoutubeDlNbwp9U0wI").at(0);

        expect(videoDiv).not.toBeNull();
        expect(videoDiv).not.toBeUndefined();
       
        mock.restore();
        done();
     })

  });


    afterAll(() => {
       // mock.restore();
    });
});
