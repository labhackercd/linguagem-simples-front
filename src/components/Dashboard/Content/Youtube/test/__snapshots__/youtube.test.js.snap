// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if youtube returns transmission not available when sessionID is null 1`] = `
<Youtube
  sessionIdDadosAbertos={null}
>
  <Styled(MuiBox)
    alignContent="center"
    display="flex"
    height="100%"
    justifyContent="center"
    width="100%"
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <WithStyles(ForwardRef(Typography))
        style={
          Object {
            "color": "grey",
          }
        }
        variant="h5"
      >
        <ForwardRef(Typography)
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          style={
            Object {
              "color": "grey",
            }
          }
          variant="h5"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
            style={
              Object {
                "color": "grey",
              }
            }
          >
             Transmissão não disponível
          </h5>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </div>
  </Styled(MuiBox)>
</Youtube>
`;
